### PR TITLE
Bug redirect after login

### PR DESCRIPTION
**Pull request by ewansheldon**

### CODE REVIEW

**Comment by LL782** [comment]
Is there an automated test that could have caught this bug/prevent it from happening again?

File: src/components/Board/Board.js [hunk 1]
```diff
@@ -29,7 +29,11 @@ const Board = ({ history, match }) => {
       setError(false);
     }).catch((err) => {
       if (err.status === 401) {
-        history.push('/login');
+        const redirect = {
```
Comment (apavlidi) [inline src/components/Board/Board.js]: Can you please extract that to a function and reuse it on the Board component as well? It seems that the if block is exactly the same.

File: src/components/Home/Home.js [hunk 2]
```diff
@@ -39,7 +39,11 @@ const Home = ({ history }) => {
       setFetching(false);
     }).catch((err) => {
       if (err.status === 401) {
-        history.push('/login');
+        const redirect = {
```
Comment (apavlidi) [inline src/components/Home/Home.js]: If you agree on refactoring here: https://github.com/codurance/Retropolis-FE/pull/47/files#r436666191 please reuse that function you extracted.

**Comment by ewansheldon** [comment]
> Is there an automated test that could have caught this bug/prevent it from happening again?

to be honest, this is less a bug that emerged from a working implementatin, and more a feature that we didn't implement when we added routing to the application. but we're currently discussing the best way to secure it with a test now

File: src/components/Login/Login.js [hunk 3]
```diff
@@ -17,7 +17,13 @@ const Login = ({ history }) => {
 
   const success = (res) => {
     successLogin(res);
-    history.push('/');
+    let url;
```
Comment (apavlidi) [inline src/components/Login/Login.js]: Same https://github.com/codurance/Retropolis-FE/pull/47/files#r436666528

**Review by apavlidi — CHANGES_REQUESTED** [review]
The problem is resolved but there are some potential refactorings.

**Review by apavlidi — APPROVED** [review]
Good job man

**Pull request merged**